### DISCUSSION TITLE
Can use a.Preferences().StringWithFallback() manipulate json data?

### DISCUSSION THREAD

**Discussion by junmaqiang** [fyne-io/fyne#3578] [Q&A] [answered]
## The data format is json, I don't want to use data software exclusively, I just want to use the one provided by fyne

```json
//var MoneyData = [][]string{
//	{"218991153800884", "2022-11-12 10:23:01", "A", "12", "20", "1"},
//	{"218991153800884", "2022-01-12 10:23:11", "B", "12", "20", "1"},
//	{"218991153800884", "2022-12-12 10:23:12", "C", "-12", "20", "2"},
//	{"218991153800884", "2022-09-12 10:13:01", "D", "-2", "20", "2"},
//}
```

## 
I see that Preferences() is close to what I thought, but the format doesn't match

**Comment by Jacalz** [comment, marked as answer]
I am not entirely sure that I understand the question. Can you not save the json data as a string and store that?

**Reply by andydotxyz** [reply]
Yeah, you can just store JSON as a string and use the Go builtin marshall/unmarshal code (the struct tags etc) to handle the data format.
